Qt5.5 support

aabadie opened the pull request:
Hi, I wanted to test the build of qtdcm against qt5.5 on windows using VS 2013. This pull request contains changes to make this work (not tested on Linux). It also contains potential leaks fixes and minor refactorings.
As a general remark, I think this code requires lots of changes in order to have it in a good shape. I'll see if I can find some time to add them.

aabadie committed:
update doxygen documentation

aabadie committed:
see #1675

aabadie committed:
testing github hook, see #1675

aabadie committed:
testing github hook, see #1675

aabadie committed:
consting things

aabadie committed:
removing useless thing

Commit:
add qt5.5 support, fix build issues on windows with vs2013, support dcmtk 3.6

Commit:
more refactoring, leak and bug fixing

Commit:
fixing build on Linux

Commit:
fixing leaks, polishing code

aabadie commented:
I pushed a few more commits that contains:
- build fix for Linux
- some leak fixes
- a small refactoring of the way preferences are handled : hopefully it fixes an issue when changing the list of available pacs)
- code improvements (pointers...)

Commit:
fixing more leaks

Commit:
still consting method, fixing issues on windows

Commit:
improving cmake code for installation

rdebroiz commented:
Which version of DCMTK did you used ? I used and this one (pretty old actually but it's still the one used by medInria) and i had some errors during compilation: 

`EXS_JPEGProcess14SV1`, `EXS_JPEGProcess1`, `EXS_JPEGProcess2_4` from `E_TransferSyntax` are not known.

it's used by `QtDcmMoveScu::acceptSubAssoc ( T_ASC_Network * aNet, T_ASC_Association ** assoc )` in a switch case.

rdebroiz commented:
> I used and this one 

 I used this one https://github.com/medInria/dcmtk

sorry

rdebroiz commented:
Ok it works with the last tagged version of DCMTK I found on the official repo, so I guess we will just use this version for medInria.

Commit:
cleaning a bit cmake files (now compatible with cmake 3)

aabadie commented:
Indeed, I use the latest release : 3.6.

Commit:
Improve several things:
- use colored and formatted logging messages
- improve windows packages generation (contains all required dlls)
- fix crash when importing series
- fix warning message when opening file dialogs
- fix temporary directories removal when quitting

aabadie commented:
I pushed ea915b2 with some minor improvements. I had a crash sometimes when downloading data from a PACS, I fixed it by changing the order of signal emission when the move completes.

aabadie commented:
Any news about this PR ?

rdebroiz commented:
I use this branch in the dev branch of medInria, I suppose that we will merge the PR as soon as the medInria dev branch will be merged in the master (with Qt5, vtk6, dtk1, etc. migration). It Should not be long.

aabadie commented:
Ok, I'll keep an eye on the medInria dev branch then. Thanks

rdebroiz commented:
you removed 
`void QtDcmLocalDicomSettingsWidget::setPreferences ( QtDcmPreferences * prefs );`

we where using it in medInria with `QtDcmPreferences::instance()` as parameter. I assume i can just remove the line since  `QtDcmPreferences::instance()` is always what is used internally ?

aabadie commented:
> I assume i can just remove the line since QtDcmPreferences::instance() is always what is used internally ?

Yes, it was a fairly bad design to pass a pointer as parameter knowing that is was possible to retrieve it internally using its static instance() method. It also makes it clearer.

Commit:
Fix QtDCMConfig.cmake.in

rdebroiz commented on CMake/QtDCMConfig.cmake.in:
```diff
@@ -2,51 +2,36 @@
 # all library dirs pour QTDCM
-SET(QTDCM_LIBRARY_DIRS  "@QTDCM_LIBRARY_DIRS@")
+set(QTDCM_LIBRARY_DIRS  "@QtDcm_LIBRARY_DIRS@")
 
```
The `QtDcm_LIBRARY_DIRS` is empty, it is not defined somwhere else in the project. the `set(QTDCM_SOURCE_DIR "@QtDcm_SOURCE_DIR@/Code")` above works because CMake automatically internally define the `${${PROJECT_NAME}_BINARY_DIR}`  and `${${PROJECT_NAME}_SOURCE_DIR}`

I don't know how you want to deal with all of it, however i made a quick fix, see https://github.com/aabadie/qtdcm/pull/4

aabadie committed:
Merge pull request #4 from rdebroiz/qt5.5_support

Fix QtDCMConfig.cmake.in

rdebroiz commented on CMake/QtDCMConfig.cmake.in:
```diff
@@ -2,51 +2,34 @@
 #path to binary dir of QTDCM
-SET(QTDCM_BINARY_DIR "@QTDCM_BIN_DIR@")
+set(QTDCM_BINARY_DIR "@QtDcm_BINARY_DIR@/lib")
 
```
My mistake here :S. I let you correct (too lazy to make a pull request just for that, and i can't push on your repo ;) )

Commit:
fixing wrong cmake variable

Commit:
fixing wrong cmake variable

aabadie commented:
Sorry, I misunderstood your mistake at first. It should be ok now.

Commit:
Add option to use FindDCMTK modules or not.

aabadie committed:
Merge pull request #5 from rdebroiz/qt5.5_support

Add option to use FindDCMTK modules or not.

aabadie commented:
@rdebroiz, if you don't mind, I made several other improvements in the current master of aabadie/qtdcm. I'd prefer to merge this branch and create another pull-request instead of continuing with this PR. What do you think ?

rdebroiz commented:
It's Okay for me.

aabadie commented:
Ok, I created  #7. Can you have a look ?

rdebroiz commented:
Ok

ocommowi committed:
Add ITK 4.9 support (linking)
